tools/print_test_stats.py

```diff
@@ -632,23 +629,22 @@ def __init__(self, name: str) -> None:
         self.total_time = 0.0
         self.test_suites: Dict[str, TestSuite] = dict()
 
-    def append(self, test_case: TestCase) -> None:
-        suite_name = test_case.class_name
+    def append(self, test_case: TestCase, test_type: str) -> None:
+        is_multi_test = self.name == 'test_cpp_extensions_aot' or \
+            self.name == 'distributed/test_distributed_fork' or \
+            self.name == 'distributed/test_distributed_spawn' or \
+            self.name == 'distributed/test_c10d_gloo' or \
+            self.name == 'cpp'  # The caffe2 cpp tests spawn duplicate test cases as well.
+        if is_multi_test:
+            suite_name = test_case.class_name + '__' + test_type
+        else:
+            suite_name = test_case.class_name
         if suite_name not in self.test_suites:
             self.test_suites[suite_name] = TestSuite(suite_name)
         if test_case.name in self.test_suites[suite_name].test_cases:
-            # We expect duplicate tests for test_cpp_extensions_aot, distributed/test_distributed_fork,
-            # and distributed/test_distributed_spawn and test_c10d_gloo.
-            # In these cases, we store the test case that took the longest,
-            # as in these jobs, the duplicate tests are run in parallel.
-            # For other unexpected cases, we should raise a warning.
-            if self.name == 'test_cpp_extensions_aot' or \
-               self.name == 'distributed/test_distributed_fork' or \
-               self.name == 'distributed/test_distributed_spawn' or \
-               self.name == 'distributed/test_c10d_gloo' or \
-               self.name == 'cpp':  # The caffe2 cpp tests spawn duplicate test cases as well.
-                time_difference = self.test_suites[suite_name].replace(test_case)
-                self.total_time += time_difference
+            if is_multi_test:
```

Comment:
here we already knew the test are ran exactly twice. so I think we are fine when the test file is in the list above